File: Book/Fortran/Fig_12.13_serialPi.f90
          PROGRAM MAIN
          IMPLICIT NONE

          INTEGER :: i
          INTEGER, PARAMETER :: num_steps = 1000000
          REAL :: x, pi, sum, step

          sum = 0.0

          step = 1.0/num_steps

          DO i = 1, num_steps
             x = (i - 0.5) * step
             sum = sum + 4.0 / (1.0 + x * x)
          ENDDO

          pi = step * sum
          print *, "pi=", pi

          END PROGRAM MAIN 

File: Book/C/Fig_12.13_serialPi.c
#include <stdio.h>
static long num_steps = 1000000;
float step;
int main()
{
   int i;
   float x, pi, sum = 0.0;

   step = 1.0f / (double) num_steps;

   for (i = 0;i < num_steps; i++) {
      x = (i + 0.5f) * step;
      sum += 4.0f / (1.0f + x * x);
   }

   pi = step * sum;
   printf("pi=%f\n",pi);
}	  
  
